coarse-grained-lock/README.md

```diff
@@ -0,0 +1,105 @@
+---
+Title: "Coarse-Grained Lock Pattern in Java: Simplifying Thread-Safe Operations"
+Short Title: Coarse-Grained Lock
+Description: "Coarse-Grained Lock pattern ensures thread safety by holding a global lock for related objects instead of multiple locks."
+Category: Concurrency
+Tags:
+
+    Concurrency
+    Synchronization
+    Thread Safety
+    Locking Mechanisms
+---
```

Comment:
Please follow the frontmatter format exactly. Here is an example from another pattern:

```
---
title: "Abstract Document Pattern in Java: Simplifying Data Handling with Flexibility"
shortTitle: Abstract Document
description: "Explore the Abstract Document design pattern in Java. Learn its intent, explanation, applicability, benefits, and see real-world examples to implement flexible and dynamic data structures."
category: Structural
language: en
tag:
  - Abstraction
  - Decoupling
  - Dynamic typing
  - Encapsulation
  - Extensibility
  - Polymorphism
---

```